src/react-components/ui-root.js

```diff
@@ -910,6 +910,7 @@ class UIRoot extends Component {
               )}
               <TwoDHUD.BottomHUD
                 onCreateObject={() => this.setState({ infoDialogType: InfoDialog.dialogTypes.create_object })}
+                onMediaPicked={AFRAME.utils.device.isMobile() ? this.handleCreateObject : null}
```

Comment:
Like I mentioned above, i would just pass n a boolean property here based on `AFRAME.utils.device.isMobile()` that configures if the add media dialog is used.